### PR TITLE
GUI ParticleComposition implementation

### PR DESCRIPTION
**Pull request by anikhalder**
Temporarily in  files MaterialPropertyController and TestMaterialPropertyController the connection for propagating signal (dataChanged and rowRemoval) between original material model and the material property's model has been disabled.

### CODE REVIEW

File: Core/DecoratedFormFactor/IFormFactorDecorator.h [hunk 1]
```diff
@@ -48,6 +48,9 @@ class BA_CORE_API_ IFormFactorDecorator : public IFormFactor
     double topZ(const IRotation& rotation) const override {
         return mp_form_factor->topZ(rotation); }
 
+    IFormFactor* getFormFactor() const {
+        return mp_form_factor; }
+
```
Comment (waltervh) [inline Core/DecoratedFormFactor/IFormFactorDecorator.h]: can you make this method return a const IFormFactor*?

File: GUI/coregui/Views/RealSpaceWidgets/TransformTo3D.cpp [hunk 2]
```diff
@@ -23,6 +23,30 @@
...
+#include "FormFactorHemiEllipsoid.h"
+#include "FormFactorIcosahedron.h"
+#include "FormFactorPrism3.h"
+#include "FormFactorPrism6.h"
+#include "FormFactorPyramid.h"
+#include "FormFactorTetrahedron.h"
+#include "FormFactorTruncatedCube.h"
+#include "FormFactorTruncatedSphere.h"
+#include "FormFactorTruncatedSpheroid.h"
+
```
Comment (waltervh) [inline GUI/coregui/Views/RealSpaceWidgets/TransformTo3D.cpp]: This can be replaced with:
#include HardParticles.h

**Comment by waltervh** [comment]
Nice work! If you can take into account my two comments, I will merge.

**Comment by anikhalder** [comment]
Thanks a lot! I implemented the changes.

**Pull request merged**